lisa/sut_orchestrator/baremetal/bootconfig.py

```diff
@@ -0,0 +1,166 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import re
+from dataclasses import dataclass, field
+from pathlib import PurePosixPath
+from typing import Optional, Type
+
+from dataclasses_json import dataclass_json
+
+from lisa import schema
+from lisa.node import quick_connect
+from lisa.tools import Cat, Sed
+from lisa.util import (
+    InitializableMixin,
+    LisaException,
+    field_metadata,
+    find_group_in_lines,
+    subclasses,
+)
+from lisa.util.logger import get_logger
+
+from .schema import BootConfigSchema
+
+BOOT_LABEL = "label lisa baremetal"
+
+
+class BootConfig(subclasses.BaseClassWithRunbookMixin, InitializableMixin):
+    def __init__(
+        self,
+        runbook: BootConfigSchema,
+    ) -> None:
+        super().__init__(runbook=runbook)
+        self.boot_config_runbook: BootConfigSchema = self.runbook
+        self._log = get_logger("boot_config", self.__class__.__name__)
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return BootConfigSchema
+
+    def config(self) -> None:
+        raise NotImplementedError()
+
+
+@dataclass_json()
+@dataclass
+class PxeBootSchema(BootConfigSchema):
+    connection: Optional[schema.RemoteNode] = field(
+        default=None, metadata=field_metadata(required=True)
+    )
+    host_name: str = field(default="", metadata=field_metadata(required=True))
+    image_source: str = field(default="", metadata=field_metadata(required=True))
+    kernel_boot_params: Optional[str] = field(default="")
+
+
+class PxeBoot(BootConfig):
+    def __init__(
+        self,
+        runbook: PxeBootSchema,
+    ) -> None:
+        super().__init__(runbook=runbook)
+        self.pxe_runbook: PxeBootSchema = self.runbook
+        self._log = get_logger("pxe_boot", self.__class__.__name__)
+        self._boot_dir = "/var/lib/tftpboot/"
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "pxe_boot"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return PxeBootSchema
+
+    def config(self) -> None:
+        assert self.pxe_runbook.connection, "connection is required for dhcp_server"
+
+        self.pxe_runbook.connection.name = "dhcp_server"
+        self._dhcp_server = quick_connect(
+            self.pxe_runbook.connection, logger_name="dhcp_server"
+        )
+
+        node_name = f"{self.pxe_runbook.host_name}"
+        pxe_boot_config_path = self._get_pxe_config_path(node_name)
+        if not pxe_boot_config_path:
+            with open(pxe_boot_config_path, "w") as f:
+                f.write("timeout 150\n\nmenu title selections\n\n")
+
+        boot_image = self.pxe_runbook.image_source.replace(
```

Comment:
Use Path.relative_to to get relative path.

Reply:
The change is done, thanks